src/main/java/mops/presentation/DirectoryController.java

```diff
@@ -0,0 +1,128 @@
+     * @param fileInfo file object
+     * @return route after completion
+     */
+    @PostMapping(path = "/{dirId}/upload")
+    public String uploadFile(KeycloakAuthenticationToken token,
+                             Model model,
+                             @PathVariable("dirId") long dirId,
+                             @Param("file") FileInfo fileInfo) {
+        final Account account = AccountUtil.getAccountFromToken(token);
+        //TODO: exception handling and user error message
+        directoryService.uploadFile(account, dirId, fileInfo);
+        return String.format("redirect:/material1/dir/%d", dirId);
+    }
+
+    /**
+     * Creates a new sub folder.
+     *
+     * @param token      keycloak auth token
+     * @param model      spring view model
+     * @param dirId      id of the parent folder
+     * @param folderName name of the new sub folder
+     * @return object of the folder
+     */
+    @PostMapping("/{dirId}/create")
+    public String createSubFolder(KeycloakAuthenticationToken token,
+                                  Model model,
+                                  @PathVariable("dirId") long dirId,
+                                  @RequestAttribute("folderName") String folderName) {
+        final Account account = AccountUtil.getAccountFromToken(token);
+        final int directoryId = directoryService.createFolder(account, dirId, folderName);
+        return String.format("redirect:/material1/dir/%d", directoryId);
+    }
+
+
+    /**
+     * Deletes a folder.
+     *
+     * @param token user credentials
+     * @param model spring view model
+     * @param dirId id of the folder to be deleted
+     * @return the id of the parent folder
+     */
+    @DeleteMapping("/{dirId}")
+    public String deleteFolder(KeycloakAuthenticationToken token,
+                               Model model,
+                               @PathVariable("dirId") long dirId) {
+        final Account account = AccountUtil.getAccountFromToken(token);
+        final int directoryId = directoryService.deleteFolder(account, dirId);
+        return String.format("redirect:/material1/dir/%d", directoryId);
+    }
+
+    /**
+     * Searches a folder for files.
+     *
+     * @param token user credentials
+     * @param model spring view model
+     * @param dirId id of the folder to be searched
+     * @param query wrapper object of the query parameter
+     * @return route to files view
+     */
+    @PostMapping("/{dirId}/search")
+    public String searchFolder(KeycloakAuthenticationToken token,
+                               Model model,
+                               @PathVariable("dirId") long dirId,
+                               @RequestAttribute("searchQuery") FileQuery query) {
+        final Account account = AccountUtil.getAccountFromToken(token);
+        final List<FileInfo> files = directoryService.searchFolder(account, dirId, query);
+        model.addAttribute("files", files);
+        return "files";
```

Comment:
don't we want this to be a specific template?
so we can show extra info.

Reply:
I'm not sure, what you meant by that. You want to do a redirect with path variables?

Reply:
its fine for now